packages/contracts-sdk/src/lib/contracts-sdk.ts

```diff
@@ -1077,18 +1078,34 @@ https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scope
     // -- go
     const mintCost = await this.pkpNftContract.read.mintCost();
 
+    let _gasLimit: BigNumberish | undefined = gasLimit;
```

Comment:
Rather than needing to duplicate all of the transaction params and also needing a local mutable variable to store the `_gasLimit` value, it would be nice to get this defined as a self-contained function -- otherwise we are likely to end up changing the tx parameters in one spot but not the other by mistake, when we really need them to always be the exact same for the estimateGas() to be accurate

```javascript
async function _mintNextAndAddAuthMethodsDynamicGasLimit({ gasLimit, mintCost, txParams }) {
  return this.pkpHelperContract.write.mintNextAndAddAuthMethods(...txParams, {
    gasLimit:
      gasLimit ||
      (await this.pkpHelperContract.write.populateTransaction.mintNextAndAddAuthMethods(
        ...txParams,
        { value: mintCost }
      )),
    value: mintCost,
  });
}

```

Reply:
Great suggestion. I  updated the generic function to receive the contract, method, params, overrides and all needed things, I believe that approach is more readable